src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDependencyDao.java

```diff
@@ -122,46 +124,46 @@ private void batchStoreSnapshotFileDependencies(
                     .whereEqualTo("snapshotId", snapshotId).get()
             );
 
-        // Scan the lookup results and launch the sets in parallel
+        // Combine the batch inputs with the results from the lookups to be used in the next batch process
         int index = 0;
-        List<ApiFuture<WriteResult>> setFutures = new ArrayList<>();
-
+        final Map<String, QuerySnapshot> fileIdToQuerySnapshot = new HashMap<>();
         for (QuerySnapshot querySnapshot : querySnapshotList) {
             String fileId = batch.get(index);
-
-            List<QueryDocumentSnapshot> documents = querySnapshot.getDocuments();
-            switch (documents.size()) {
-                case 0: {
-                    // no dependency yet. Let's make one
-                    FireStoreDependency fireStoreDependency = new FireStoreDependency()
-                        .snapshotId(snapshotId)
-                        .fileId(fileId)
-                        .refCount(1L);
-                    DocumentReference docRef = depColl.document();
-                    setFutures.add(docRef.set(fireStoreDependency));
-                    break;
-                }
-
-                case 1: {
-                    // existing dependency; increment the reference count
-                    QueryDocumentSnapshot docSnap = documents.get(0);
-                    FireStoreDependency fireStoreDependency = docSnap.toObject(FireStoreDependency.class);
-                    fireStoreDependency.refCount(fireStoreDependency.getRefCount() + 1);
-                    setFutures.add(docSnap.getReference().set(fireStoreDependency));
-                    break;
-                }
-
-                default:
-                    throw new FileSystemCorruptException(
-                        "Found more than one document for a file dependency - fileId: " + fileId);
-            }
+            fileIdToQuerySnapshot.put(fileId, querySnapshot);
             index++;
         }
 
         // Collect the set results
         fireStoreUtils.batchOperation(
-            setFutures,
-            future -> future
+            Lists.newArrayList(fileIdToQuerySnapshot.entrySet()),
```

Comment:
Why is this better than `new ArrayList<>()`? I think this usage was obsoleted by the diamond operator in java 7.
```suggestion
            new ArrayList<>(fileIdToQuerySnapshot.entrySet()),
```

Reply:
I tried new ArrayList but the factory was a little easier to use (I was having trouble with generics in the lambda below not recognizing that it was an Entry

Reply:
What error are you seeing? I tried making this change on your branch locally and it worked fine for me.